toolkit/tools/imagegen/diskutils/diskutils.go

```diff
@@ -620,14 +620,22 @@ func InitializeSinglePartition(diskDevPath string, partitionNumber int, partitio
 
 	logger.Log.Debugf("Initializing partition device path: %v", partDevPath)
 
-	// Set partition friendly name (only for gpt)
+	// Set partition friendly name and partition type UUID (only for gpt)
 	if partitionTableType == "gpt" {
 		partitionName := partition.Name
 		_, stderr, err := shell.Execute("flock", "--timeout", timeoutInSeconds, diskDevPath, "parted", diskDevPath, "--script", "name", partitionNumberStr, partitionName)
 		if err != nil {
 			logger.Log.Warnf("Failed to set partition friendly name using parted: %v", stderr)
 			// Not-fatal
 		}
+
+		if partition.TypeUUID != "" {
```

Comment:
Just do double check, is this going to generate a reasonable error on old parted if the `type` parameter is not supported, or will it be some cryptic undecipherable error?

(for background 2.0's version of parted is too old to support this if I recall right)

Reply:
We could also consider adding support in `imageconfigvalidator` that makes sure you have a new enough version of `parted` if you set custom values for the type.

Reply:
`parted` will just print out its help string to stdout and return exit code 0. The type parameter is effectively ignored.